### PR TITLE
Change mobile nav to be full height and fixed

### PR DESCRIPTION
**Pull request by joe-hastings**
Don't display the nav under the header instead make it full height and move the trigger on top. Also, make the burger menu use a CSS variable for its size rather than being hard-coded.

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js [hunk 1]
```diff
@@ -20,34 +20,36 @@ export default class Navigation {
...
-      `.nav-Header_Items,
-      .nav-Header_Backdrop,
-      .nav-Header_Dropdown,
-      .nav-Mobile`
-    )
   }
 
+  /**
+   * If we're on mobile we want to track the selected states to control the sub navigation open
+   * / closed. aria-selected won't be used on lg++ so we programmatically add it just for sizes
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js]: We shouldn't be using `aria-` attributes at all; the entire mobile nav is `aria-hidden="true"` so it makes no sense.

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/base.html [hunk 2]
```diff
@@ -1,3 +1,5 @@
+{% import 'components/macros/components.html' as components %}
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/base.html]: This should not be in the `components` app; it's site-wide and on every site so it shouldn't be in an app folder.

**Review by lewiscollard — CHANGES_REQUESTED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js [hunk 3]
```diff
@@ -20,34 +20,36 @@ export default class Navigation {
     this.setupListeners()
   }
 
+  /**
+   * Cache any dom elements we will use multiple times.
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js]: DOM*

**Review by lewiscollard — CHANGES_REQUESTED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/css/globals/navigation/mobile.css [hunk 4]
```diff
@@ -162,17 +185,19 @@
 */
 .nav-Mobile_Dropdown,
 .nav-Mobile_DropdownChild {
-  position: absolute;
-  top: 0;
+  position: fixed;
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/css/globals/navigation/mobile.css]: I think I see a potential to break, though I haven't tested it: if you have a third level of navigation that has a transform on the parent (e.g. you're transitioning it in) won't `position: fixed` create a new stacking context which would then make the third level positioned `var(--Mobile_NavHeight)` from the height of its parent and *not* from the document?

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js (hunk 1 above)
Comment (joe-hastings) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js]: It get's used for the styling of the current active items, as it's also `aria-selected` for desktop it kind of made sense to me to just leave it done like this for mobile too, I could replace with a modifier class for handling active items for mobile though if you'd prefer?

Comment (joe-hastings) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/js/navigation/navigation.js]: Actually my bad was getting aria-current and aria-selected mixed up ignore the above 😁

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/base.html (hunk 2 above)
Comment (joe-hastings) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/base.html]: So should the macro go in `apps/site/templates/site/macros/site.html`?

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/css/globals/navigation/mobile.css (hunk 4 above)
Comment (joe-hastings) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/assets/css/globals/navigation/mobile.css]: Nice catch, tricky one to fix 🤔

**Review by lewiscollard — APPROVED** [review]

**Pull request merged**